File: c#/shapes/Rectangle.cs
namespace Canvas;

public class Rectangle : Shape
{
    public int Width { get; set; }
    public int Height { get; set; }

    public Rectangle(int width, int height)
    {
        Type = "Rectangle";
        Width = width;
        Height = height;
    }

    public override int CalArea(int borderThickness)
    {
        return (Width + 2 * borderThickness) * (Height + 2 * borderThickness);
    }

}

File: java/src/shapes/Rectangle.java
package shapes;

public class Rectangle extends Shape {
    public int width;
    public int height;

    public Rectangle(int width, int height, int borderThickness) {
        this.width = width;
        this.height = height;
        this.borderThickness = borderThickness;
    }

    public int calculateArea(boolean withBorder) {
        if (withBorder) {
            return (width + borderThickness) * (height + borderThickness);
        }
        return width * height;
    }
}
